api/src/main/java/org/pmiops/workbench/cohortreview/AnnotationQueryBuilder.java

```diff
@@ -0,0 +1,219 @@
+        if (definition == null) {
+          throw new BadRequestException("Invalid annotation name: " + column);
+        }
+        annotationCount++;
+        fromBuilder.append(
+            String.format(ANNOTATION_JOIN_SQL, annotationCount, annotationCount, annotationCount,
+                definition.getCohortAnnotationDefinitionId(), cohortReview.getCohortReviewId()));
+        String sourceColumn;
+        if (definition.getAnnotationType().equals(AnnotationType.ENUM)) {
+          sourceColumn = String.format("ae%d.name", annotationCount);
+          fromBuilder.append(
+              String.format(ANNOTATION_VALUE_JOIN_SQL, annotationCount, annotationCount, annotationCount));
+        } else {
+
+          String columnName = ANNOTATION_COLUMN_MAP.get(definition.getAnnotationType());
+          if (columnName == null) {
+            throw new BadRequestException("Invalid annotation type: " + definition.getAnnotationType());
+          }
+          sourceColumn = String.format("a%d.%s", annotationCount, columnName);
+        }
+        String columnAliasName = String.format("av%d", annotationCount);
+        selectBuilder.append(String.format("%s %s", sourceColumn, columnAliasName));
+        columnAliasMap.put(column, columnAliasName);
+      }
+    }
+    return selectBuilder.toString() + fromBuilder.toString();
+  }
+
+  private String getWhereSql(CohortReview cohortReview, List<CohortStatus> statusFilter) {
+    StringBuilder whereBuilder = new StringBuilder(
+        String.format("\nWHERE pcs.cohort_review_id = %d", cohortReview.getCohortReviewId()));
+    if (!statusFilter.containsAll(REVIEWED_STATUSES)) {
+      whereBuilder.append(" AND pcs.status IN (");
+      whereBuilder.append(Joiner.on(", ").join(statusFilter.stream().map(CohortStatus::ordinal).toArray()));
+      whereBuilder.append(")");
+    }
+    return whereBuilder.toString();
+  }
+
+  private String getOrderBySql(List<String> orderBy, Map<String, String> columnAliasMap) {
+    StringBuilder orderByBuilder = new StringBuilder("\nORDER BY ");
+    boolean firstOrderByColumn = true;
+    for (String orderByColumn : orderBy) {
+      if (firstOrderByColumn) {
+        firstOrderByColumn = false;
+      } else {
+        orderByBuilder.append(", ");
+      }
+      if (orderByColumn.equals(PERSON_ID_COLUMN) || orderByColumn.equals(REVIEW_STATUS_COLUMN)) {
+        orderByBuilder.append(orderByColumn);
+      } else {
+        String columnAlias = columnAliasMap.get(orderByColumn);
+        if (columnAlias == null) {
+          throw new BadRequestException("Column " + orderByColumn + " in orderBy must be present in columns");
+        }
+        orderByBuilder.append(columnAlias);
+      }
+    }
+    return orderByBuilder.toString();
+  }
+
+  private String getLimitAndOffsetSql(int limit, long offset) {
+    StringBuilder endSqlBuilder = new StringBuilder("\nLIMIT ");
+    endSqlBuilder.append(limit);
+    if (offset != 0L) {
+      endSqlBuilder.append(" OFFSET ");
```

Comment:
Fine for initial impl, but should be a TODO here to use a more robust implementation, as cohort annotations are mutable.

Reply:
Done.